src/main.nr

```diff
@@ -7,7 +9,17 @@ fn main(value: u32, hash_value: pub u32){
 }
 
 fn calculate_hash(num: u32) -> u32 {
-    
+    let num_field = num as Field;
+    let num_bytes = num_field.to_be_bytes(4);
+    let size = 2.pow_32(32) as u64;
+    let mut hash_value: u64 = 0x811C9DC5;
+
+    for i in 0..4 {
+        let product = (hash_value * (PRIME as u64)) % size;
```

Comment:
To avoid this mod operation, you can do something like this so that Noir wraps the value to 32 bits itself

```
fn calculate_hash(num: u32) -> u32 {
    let num_field = num as Field;
    let num_bytes = num_field.to_be_bytes(4);
    let size = 2.pow_32(32) as u64;
    let mut hash_value: u32 = 0x811C9DC5;

    for i in 0..4 {
        let product = hash_value * PRIME;
        hash_value = (product ^ (num_bytes[i] as u32));
    }

    hash_value as u32
}
```